### PR TITLE
Added scm integrations types

### LINKED ISSUE #123: Add SCM integrations types
Add SCM integrations types constants to the project, for use it in the `CLI`.

```
type ScmType string

// Type of SCM integrations
const (
	ScmGithubType           ScmType = "github"             // GitHub
	ScmGithubEnterpriseType ScmType = "github-enterprise"  // GitHub Enterprise (private instance)
	ScmGitlabType           ScmType = "gitlab"             // GitLab.com
	ScmGitlabSelfHostedType ScmType = "gitlab-self-hosted" // GitLab self-hosted (private instance)
	ScmBitbucketType        ScmType = "bitbucket"          // Bitbucket
)

```

### PR DESCRIPTION
**Pull request by brandon-welsch**
Close #123

### CODE REVIEW

File: integrations.go [hunk 1]
```diff
@@ -6,6 +6,16 @@ import (
 	"github.com/Scalingo/go-scalingo/http"
 )
 
+type ScmType string
```
Comment (EtienneM) [inline integrations.go]:
```suggestion
type SCMType string
```

Comment (brandon-welsch) [reply integrations.go]: Your review may be a pain in the ass, but it's relevant to a lot of small points. And that's not so bad :)

File: integrations.go [hunk 2]
```diff
@@ -6,6 +6,16 @@ import (
 	"github.com/Scalingo/go-scalingo/http"
 )
 
+type SCMType string
+
+// Type of SCM integrations
+const (
+	ScmGithubType           ScmType = "github"             // GitHub
```
Comment (EtienneM) [inline integrations.go]: Well, then you need to update all these constants name and type ;)

File: integrations.go [hunk 3]
```diff
@@ -6,6 +6,16 @@ import (
 	"github.com/Scalingo/go-scalingo/http"
 )
 
+type SCMType string
+
+// Type of SCM integrations
+const (
+	ScmGithubType           SCMType = "github"             // GitHub
```
Comment (EtienneM) [inline integrations.go]:
```suggestion
	SCMGithubType           SCMType = "github"             // GitHub
```
Hum...

Comment (brandon-welsch) [reply integrations.go]: Sorry ^^

**Pull request merged**